config/filter.d/asterisk.conf

```diff
@@ -19,7 +19,7 @@ iso8601 = \d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+[+-]\d{4}
 log_prefix= (?:NOTICE|SECURITY)%(__pid_re)s:?(?:\[C-[\da-f]*\])? \S+:\d*( in \w+:)?
 
 failregex = ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s Registration from '[^']*' failed for '<HOST>(:\d+)?' - (Wrong password|Username/auth name mismatch|No matching peer found|Not a local domain|Device does not match ACL|Peer is not supposed to register|ACL error \(permit/deny\)|Not a local domain)$
-            ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s Call from '[^']*' \(<HOST>:\d+\) to extension '\d+' rejected because extension not found in context 'default'\.$
+            ^(%(__prefix_line)s|\[\]\s*)%(log_prefix)s Call from '[^']*' \(<HOST>:\d+\) to extension '[^']*' rejected because extension not found in context
```

Comment:
0\. no matter (because the info at end has no meaning) and it is already anchored at the begin
1 and 2. No, because in since of current failure, all this messages are the same (just failure X):
```
1.2.3.4 failure because reason X
1.2.3.4 failure because reason X, protocol=tcp
1.2.3.4 failure because reason X, whatever
```
3\. that may be possible by real PCRE (finite automata like NFA, TCL, thompson etc.) not for regexp engine of python...
Here is an example of profiler:
```
   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000    2.521    2.521 misctestcase.py:175(_testAnchorPosLeftRight)
        1    0.000    0.000    2.204    2.204 misctestcase.py:170(_testAnchorPosLeft)
        1    0.000    0.000    1.326    1.326 misctestcase.py:172(_testAnchorNegLeft)
        1    0.000    0.000    1.326    1.326 misctestcase.py:177(_testAnchorNegLeftRight)
```

```python
class TestsUtilsTest(unittest.TestCase):

	l = [
		"NOTICE[3477][C-000003bb] chan_sip.c: Call from '' (1.2.3.4:10836) to extension '++441772285411' rejected because extension not found in context 'default'.",
		"NOTICE[3477][C-000003bb] chan_sip.c: just something else",
	]
	r = [
		r"^\w+\[\d+\]\[[\w-]+\] chan_sip.c: Call from '[^']*' \(<HOST>:\d+\) to extension '[^']*' rejected because extension not found",
		r"^\w+\[\d+\]\[[\w-]+\] chan_sip.c: Call from '[^']*' \(<HOST>:\d+\) to extension '[^']*' rejected because extension not found in context '[^']*'\.$",
	]

	def _testAnchor(self, r, l):
		r = r.replace("<HOST>", "(?:::f{4,6}:)?(?P<host>[\w\-.^_]*\w)")
		r = re.compile(r, re.MULTILINE)
		for i in xrange(1000000):
			r.search(l+'\n')
		return r.search(l+'\n')

	def _testAnchorPosLeft(self):
		self.assertTrue( self._testAnchor(TestsUtilsTest.r[0], TestsUtilsTest.l[0]))
	def _testAnchorNegLeft(self):
		self.assertFalse(self._testAnchor(TestsUtilsTest.r[0], TestsUtilsTest.l[1]))

	def _testAnchorPosLeftRight(self):
		self.assertTrue( self._testAnchor(TestsUtilsTest.r[1], TestsUtilsTest.l[0]))
	def _testAnchorNegLeftRight(self):
		self.assertFalse(self._testAnchor(TestsUtilsTest.r[1], TestsUtilsTest.l[1]))

	def testAnchor(self):
		self._testAnchorPosLeft()
		self._testAnchorPosLeftRight()
		self._testAnchorNegLeft()
		self._testAnchorNegLeftRight()
```